app/src/main/java/com/github/se/travelpouch/model/notifications/NotificationRepositoryFirestore.kt

```diff
@@ -0,0 +1,153 @@
+
+/**
+ * Repository class for managing notifications in the Firestore database.
+ *
+ * @property firestore The Firestore instance used for database operations.
+ */
+class NotificationRepositoryFirestore(private val firestore: FirebaseFirestore) :
+    NotificationRepository {
+
+  // Reference to the "notifications" collection in Firestore
+  private val notificationCollection = firestore.collection("notifications")
+
+  override fun getNewUid(): String {
+    return notificationCollection.document().id
+  }
+
+  /**
+   * Adds a new notification to the Firestore database.
+   *
+   * @param notification The notification to be added.
+   */
+  override fun addNotification(notification: Notification) {
+    notificationCollection
+        .document(notification.notificationUid)
+        .set(notification)
+        .addOnSuccessListener { Log.d("NotificationRepository", "Notification added successfully") }
+        .addOnFailureListener { e ->
+          Log.e("NotificationRepository", "Error adding notification", e)
+        }
+  }
+
+  /**
+   * Fetches notifications for a specific user from the Firestore database.
+   *
+   * @param userId The UID of the user whose notifications are to be fetched.
+   * @param onNotificationFetched Callback function to be invoked with the list of fetched
+   *   notifications.
+   */
+  override fun fetchNotificationsForUser(
+      userId: String,
+      onNotificationFetched: (List<Notification>) -> Unit
+  ) {
+    notificationCollection
+        .whereEqualTo("receiverUid", userId)
+        .orderBy("timestamp", Query.Direction.DESCENDING)
+        .get()
+        .addOnCompleteListener { task ->
+          if (task.isSuccessful) {
+            val notifications = task.result?.documents?.map { documentToNotification(it) }
+            onNotificationFetched(notifications ?: emptyList())
+          } else {
+            Log.e("NotificationRepository", "Error fetching notifications", task.exception)
+          }
+        }
+        .addOnFailureListener { e ->
+          Log.e("NotificationRepository", "Error fetching notifications", e)
+        }
+  }
+
+  /**
+   * Marks a notification as read in the Firestore database.
+   *
+   * @param notificationUid The UID of the notification to be marked as read.
+   */
+  override fun markNotificationAsRead(notificationUid: String) {
+    notificationCollection.document(notificationUid).update("status", NotificationStatus.READ)
+  }
+
+  override fun deleteAllNotificationsForUser(
+      userUid: String,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    notificationCollection
+        .whereEqualTo("receiverUid", userUid)
+        .get()
+        .addOnSuccessListener { documents ->
+          for (document in documents) {
+            notificationCollection.document(document.id).delete()
+          }
+          onSuccess()
+        }
+        .addOnFailureListener { exception -> onFailure(exception) }
+  }
```

Comment:
I have some doubts concerning this function. You are not testing if the delete operation is indeed successful or not. You may need to add a firebase listener to track for the success of the operation and log any error. Also, it may be better to bnatch this operation. What you could do is directly call delete and not get, since you only need the receiverUid.

Reply:
PS : as a follow-up we cannot delete by batch, since we first need to get the document then delete it. 
https://stackoverflow.com/questions/47180076/how-to-delete-document-from-firestore-using-where-clause